lib/internal/modules/cjs/loader.js

```diff
@@ -226,6 +228,23 @@ ObjectDefineProperty(Module, 'wrapper', {
   }
 });
 
+let warned = false;
+ObjectDefineProperty(Module.prototype, 'parent', {
+  get: function() {
+    const parent = moduleParentCache.get(this);
+    if (!process.noDeprecation && !warned && parent == null) {
+      warned = true;
+      process.emitWarning(
+        'module.parent is deprecated and may be removed in a later ' +
+          'version of Node.js.',
```

Comment:
I would be more explicit about what this warns against – something stating that `module.parent` may not always be accurate. Removing it is something that I don’t hope anybody’s suggesting at this point?

Reply:
I would bet that this will never be removed (probably ever), I just copied another deprecation warning and changed the function name. Would you be OK with the following phrasing?

```suggestion
        'module.parent is deprecated and is know to break when using ' +
          'ECMAScript module specification. Consider replacing it with ' +
          'require.main === module.',
```